illwill.nim

```diff
@@ -513,16 +711,45 @@ proc exitFullScreen() =
     eraseScreen()
     setCursorPos(0, 0)
 
-proc illwillInit*(fullScreen: bool = true) =
+when defined(posix):
+  proc enableMouse() =
+    stdout.write mouseTrackAny
+    stdout.flushFile()
+
+  proc disableMouse() =
+    stdout.write disableMouseTrackAny
+    stdout.flushFile()
+else:
+  proc enableMouse(hConsoleInput: Handle) =
+    var currentMode: DWORD
+    discard getConsoleMode(hConsoleInput, addr(currentMode))
+    discard setConsoleMode(hConsoleInput,
+      ENABLE_WINDOW_INPUT or ENABLE_MOUSE_INPUT or ENABLE_EXTENDED_FLAGS or
+      (currentMode and ENABLE_QUICK_EDIT_MODE.bitnot())
+    )
+
+  proc disableMouse(hConsoleInput: Handle, oldConsoleMode: DWORD) =
+    discard setConsoleMode(hConsoleInput,  oldConsoleMode) # TODO: REMOVE MOUSE OPTION ONLY?
```

Comment:
Remove extra spaces, TODO is still relevant?

Reply:
yeah kinda, i'm not yet super happy with the windows mode change, we could maybe just set the old consoleMode (that you've already saved) then the logic would be simpler

Reply:
That sounds better, yes.